pkg/kubelet/metrics/metrics.go

```diff
@@ -412,12 +412,13 @@ var (
 		[]string{"runtime_handler"},
 	)
 
-	// RunningPodCount is a gauge that tracks the number of Pods currently running
+	// RunningPodCount is a gauge that tracks the number of Pods currently with a running pause/sandbox container
+	// It is used to expose the kubelet internal state: how many pods have running containers in the container runtime, and mainly for debugging purpose.
 	RunningPodCount = metrics.NewGauge(
 		&metrics.GaugeOpts{
 			Subsystem:      KubeletSubsystem,
 			Name:           RunningPodsKey,
-			Help:           "Number of pods currently running",
+			Help:           "Number of pods currently with a running pause container",
```

Comment:
The pause container is an implementation detail.  Containers may have running pod sandboxes with no corresponding pause container on the host.  For example, CRI-O avoids creation of a pause container when specific resources on the pod spec do not need to be shared among containers in the pod.

Can we update this text to say:

"Number of pods that have a running pod sandbox"

Thanks!

Reply:
Thanks for the detailed explanation. It makes sense.

Updated. @derekwaynecarr